src/rust/engine/src/context.rs

```diff
@@ -118,14 +119,20 @@ impl Core {
         remoting_opts.store_timeout_multiplier,
         remoting_opts.store_maximum_timeout,
       )?;
-
+      let mut store_headers = remoting_opts.store_headers.clone();
+      if let Some(oauth_bearer_token) = oauth_bearer_token {
+        store_headers.insert(
+          "authorization".to_owned(),
+          format!("Bearer {}", oauth_bearer_token.trim()),
+        );
+      }
```

Comment:
We're using this logic that adds an authorization header in a few places - does it make sense to break it out into a separate function?

Reply:
I don't think so. We only have it twice in Rust land now, for the `fs_util` and `process_executor` programs.

We used to have it in 3 places in context.rs, but that's fixed in the last commit by abstracting up to Python.